modules/wizard/view-wizard-step-home-page.php

```diff
@@ -5,29 +5,21 @@
  * @package Polylang
  *
  * @since 2.7
+ *
+ * @var PLL_Model      $model                  `PLL_Model` instance.
+ * @var WP_Post        $home_page              Home page defined in WordPress options.
+ * @var PLL_Language   $home_page_language     Home page language if already assigned.
+ * @var PLL_Language   $default_language       Polylang default language option.
+ * @var int[]          $translations           Ids of home page translations.
+ * @var PLL_Language[] $untranslated_languages List of languages in which the home page isn't translated.
  */
 
-if ( ! defined( 'ABSPATH' ) ) {
-	exit; // Don't access directly.
-}
-
-$languages = $this->model->get_languages_list();
-$default_language = count( $languages ) > 0 ? $this->options['default_lang'] : null;
-$home_page_id = get_option( 'page_on_front' );
-$translations = $this->model->post->get_translations( $home_page_id );
-$untranslated_languages = array();
-$home_page = $home_page_id > 0 ? get_post( $home_page_id ) : null;
-$home_page_language = $this->model->post->get_language( $home_page_id );
+defined( 'ABSPATH' ) || exit;
 
-foreach ( $languages as $language ) {
-	if ( ! $this->model->post->get( $home_page_id, $language ) ) {
-		$untranslated_languages[] = $language;
-	}
-}
 ?>
-<input type="hidden" name="home_page" value="<?php echo esc_attr( $home_page->ID ); ?>" />
+<input type="hidden" name="home_page" value="<?php echo esc_attr( (string) $home_page->ID ); ?>" />
 <input type="hidden" name="home_page_title" value="<?php echo esc_attr( $home_page->post_title ); ?>" />
-<?php if ( false !== $home_page_language ) : ?>
+<?php if ( ! empty( $home_page_language ) ) : ?>
```

Comment:
Note sure that we should test this within the view. If everything went smoothly, this is defined and not empty. If something went wrong, displaying half of the view may just make things worse. 

The same for the default language. 

Maybe we could make tests before we add the  step.

Reply:
Normally, everything goes smoothly because the step before this one is the assignment of a language to the whole content including the home page.

I'm going to see the PHPStan behaviour when I code an early return in the method as I did for the `$home_page`
https://github.com/polylang/polylang/blob/da9ac88efbdd7cd942872833cb3ba9d74b49fd36/modules/wizard/wizard.php#L770-L772

I think it's a edge case if a user remind the URL of this wizard step and start at this point when Polylang isn't yet properly set up.

Reply:
Done in a2a3184aa0d7e724d30c60e41f6c799aaafea106